api/src/api/post/post.services.ts

```diff
@@ -64,13 +64,32 @@ const updatePost = async (
   content: string,
   postPhotos: UploadedPhoto[]
 ) => {
-  postPhotos.forEach((photo) => {
+  // for each photo use prisma client and delete them, then recreate them, easier than
+  // doing multiple put requests, we can optimize later. Just delete all, and then create all.
+  // front-end needs to change for this as well, no filtering is required.
+
+  // Get all the images
+  const currentPost = await getPost(postId);
+
+  // Deleting
+  await currentPost?.postPhotos.forEach(async (photo) => {
     if (photo.id) {
-      updatePostPhoto(photo.url, photo.id);
-    } else {
-      createPostPhoto(photo.url, postId);
+      await deletePostPhoto(photo.id);
     }
   });
+
+  // Add all of the photos back to the database
+  await postPhotos.forEach((photo) => {
+    createPostPhoto(photo.url, postId);
+  });
+
+  // postPhotos.forEach((photo) => {
+  //   if (photo.id) {
+  //     updatePostPhoto(photo.url, photo.id);
+  //   } else {
+  //     createPostPhoto(photo.url, postId);
+  //   }
+  // });
```

Comment:
This previous logic is how it should work. The frontend when updating posts should send the following:
```js
{
  ...
  postPhotos: [
    {id: 1, url: "something.jpg"},   // This is an existing photos
    {url: "something1.jpg"},   // This is a new photo
  ],
  ...
}
```

The logic you changed already took that into account. Not sure what you meant by "doing multiple put requests" cuz there is no need to do that.